src/Compilers/Core/Portable/ConstantValueSpecialized.cs

```diff
@@ -560,6 +560,22 @@ public override sbyte SByteValue
                 }
             }
 
+            public override short Int16Value
```

Comment:
Actually, no. I changed my mind :-)
However intuitive it might be to return same 255, it is not what C# constant conversion does, and it  might be more consistent to be consistent with that:

I.E. UInt16Value should have same behavior as in:

            System.Console.WriteLine(unchecked((ushort)sbyte.MinValue));
            System.Console.WriteLine(unchecked((ushort)sbyte.MaxValue));

            System.Console.WriteLine(unchecked((ushort)byte.MinValue));
            System.Console.WriteLine(unchecked((ushort)byte.MaxValue));

That would imply sign-extending when underlying is a signed value according to the discriminant.

Then we truly can, one day, use this in constant folding.

---
In reply to: [110450281](https://github.com/dotnet/roslyn/pull/18521#discussion_r110450281) [](ancestors = 110450281,110446716)